src/Type/RecursionGuard.php

```diff
@@ -8,6 +8,12 @@ class RecursionGuard
 	/** @var true[] */
 	private static $context = [];
 
+	/**
+	 * @param Type $type
+	 * @param callable(): Type $callback
```

Comment:
Is there a space too much?

```suggestion
	 * @param callable():Type $callback
```

Reply:
No, there's only 1 space. This is consistent with everywhere else in the project.

Reply:
I see. I didnt  know that phpdocs are allowed to contain spaces